src/vehicle.cpp

```diff
@@ -3625,6 +3689,7 @@ void vehicle::consume_fuel( double load = 1.0 )
                 } else {
                     amnt -= parts[elem].amount;
                     parts[elem].amount = 0;
+                    disable_devices( ft.id );
```

Comment:
This branch is taken when a *single* tank part on the vehicle is emptied (or already empty), but there might be more tanks that can supply the fuel. The chainsaws don't have their own tank and should work as long as there is some fuel in any of the tanks.